packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py

```diff
@@ -326,42 +388,127 @@ def build(self):
 
 
-class BoxesFromPointsTaskBuilder:
+class PointsTaskBuilder(SimpleTaskBuilder):
     def __init__(self, manifest: TaskManifest, escrow_address: str, chain_id: int) -> None:
-        self.exit_stack = ExitStack()
-        self.manifest = manifest
-        self.escrow_address = escrow_address
-        self.chain_id = chain_id
+        super().__init__(manifest=manifest, escrow_address=escrow_address, chain_id=chain_id)
 
-        self.logger: Logger = NullLogger()
+        self._mean_gt_bbox_radius_estimation: _MaybeUnset[float] = _unset
+
+    def _parse_gt_dataset(self, gt_file_data, *, add_prefix=None):
+        gt_dataset = super()._parse_gt_dataset(gt_file_data, add_prefix=add_prefix)
+
+        assert len(gt_dataset.categories()[dm.AnnotationType.label]) == 1
+        label_id = 0
+
+        updated_gt_dataset = dm.Dataset(categories=gt_dataset.categories(), media_type=dm.Image)
+
+        # Replace boxes with their center points
+        # Collect radius statistics
+        radiuses = []
+        for gt_sample in gt_dataset:
+            image_size = gt_sample.media_as(dm.Image).size
+            image_half_diag = ((image_size[0] ** 2 + image_size[1] ** 2) ** 0.5) / 2
+
+            sample_points = []
+            for gt_bbox in gt_sample.annotations:
+                if not isinstance(gt_bbox, dm.Bbox):
+                    continue
+
+                x, y, w, h = gt_bbox.get_bbox()
+                bbox_center = dm.Points([x + w / 2, y + h / 2], label=gt_bbox.label, id=gt_bbox.id)
+
+                rel_int_bbox_radius = min(w, h) / 2 / image_half_diag
+                radiuses.append(rel_int_bbox_radius)
+
+                sample_points.extend(bbox_center.points)
+
+            # Join points into a single annotation for compatibility with CVAT capabilities
+            updated_anns = [dm.Points(sample_points, label=label_id)]
+
+            updated_gt_dataset.put(gt_sample.wrap(annotations=updated_anns))
+
+        self._mean_gt_bbox_radius_estimation = min(0.5, np.mean(radiuses).item())
+
+        return updated_gt_dataset
+
+    def _setup_gt_job_for_cvat_task(
+        self, task_id: int, gt_dataset: dm.Dataset, *, dm_export_format: str = "datumaro"
+    ) -> None:
+        super()._setup_gt_job_for_cvat_task(
+            task_id=task_id, gt_dataset=gt_dataset, dm_export_format=dm_export_format
+        )
+
+    def _setup_quality_settings(self, task_id: int, **overrides) -> None:
+        assert self._mean_gt_bbox_radius_estimation is not _unset
+
+        values = {
+            # We have at most 1 annotation per frame, so accuracy is either 0 or 1.
+            # Point set accuracy is controlled by iou_threshold,
+            # so configure it with the requested value.
+            "target_metric_threshold": 0.9,
+            "iou_threshold": self.manifest.validation.min_quality,
+            "oks_sigma": self._mean_gt_bbox_radius_estimation,
+            "use_bbox_size_for_points": False,
```

Comment:
```suggestion
            "point_size_base": "image_size",
```

https://github.com/cvat-ai/cvat/pull/8634